modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java

```diff
@@ -1229,9 +1228,20 @@ private static void checkTableWithData(Ignite ignite, String name) {
         assertNotNull(table);
 
         for (int i = 0; i < 100; i++) {
-            Tuple row = table.keyValueView().get(null, Tuple.create().set("id", i));
+            Tuple row;
 
-            assertEquals(VALUE_PRODUCER.apply(i), row.stringValue("name"));
+            try {
+                row = table.keyValueView().get(null, Tuple.create().set("id", i));
+
+                assertEquals(VALUE_PRODUCER.apply(i), row.stringValue("name"));
+            } catch (TransactionException te) {
+                try {
+                    // There may be an exception if the primary replica node was stopped. We should wait for a new primary to appear.
+                    Thread.sleep(100);
```

Comment:
Does it mean that we just skip a row and move on to the next one? Is it correct test behavior?

Reply:
My bad, missed retry code.